learning/prompts/documentation-lookup/44_dataflow_runner.md

```diff
@@ -0,0 +1,85 @@
+
+To execute Apache Beam pipelines using the Cloud Dataflow Runner, follow these steps:
+
+***1. Setup Your Cloud Project and Resources:***
+
+Complete the steps outlined in the 'Before You Begin' section of the [Cloud Dataflow quickstart](https://cloud.google.com/dataflow/docs/quickstarts) for your chosen programming language.
+1. Select or create a Google Cloud Platform Console project.
+2. Enable billing for your project.
+3. Enable the required Google Cloud APIs, including Cloud Dataflow, Compute Engine, Stackdriver Logging, Cloud Storage, Cloud Storage JSON, and Cloud Resource Manager. Additional APIs may be necessary depending on your pipeline code.
+4. Authenticate with Google Cloud Platform.
+5. Install the Google Cloud SDK.
+6. Create a Cloud Storage bucket.
+
+***2. Specify Dependencies (Java Only):***
+
+When using the Apache Beam Java SDK, specify your dependency on the Cloud Dataflow Runner in the `pom.xml` file of your Java project directory.
+
+```java
+<dependency>
+  <groupId>org.apache.beam</groupId>
+  <artifactId>beam-runners-google-cloud-dataflow-java</artifactId>
+  <version>2.54.0</version>
+  <scope>runtime</scope>
+</dependency>
+```
+
+Ensure that you include all necessary dependencies to create a self-contained application. In some cases, such as when starting a pipeline using a scheduler, you'll need to package a self-executing JAR by explicitly adding a dependency in the Project section of your `pom.xml` file. For more details about running self-executing JARs on Cloud Dataflow, refer to the [Self-executing JAR](https://beam.apache.org/documentation/runners/dataflow/#self-executing-jar) section in the Apache Beam documentation on Cloud Dataflow Runner.
+
+***3. Configure Pipeline Options:***
+
+Configure the execution details, including the runner (set to `dataflow` or `DataflowRunner`), Cloud project ID, region, and streaming mode, using the [`PipelineOptions`](https://beam.apache.org/releases/pydoc/current/apache_beam.options.pipeline_options.html#apache_beam.options.pipeline_options.PipelineOptions) interface for Python or the [`DataflowPipelineOptions`](https://beam.apache.org/releases/javadoc/current/index.html?org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.html) interface for Java.
+
+You can utilize pipeline options to control various aspects of how Cloud Dataflow executes your job. For instance, you can specify whether your pipeline runs on worker virtual machines, on the Cloud Dataflow service backend, or locally. For additional pipeline configuration options, refer to the reference documentation for the [`PipelineOptions`](https://beam.apache.org/releases/pydoc/current/apache_beam.options.pipeline_options.html#apache_beam.options.pipeline_options.PipelineOptions) interface (Python) or the [`DataflowPipelineOptions`](https://beam.apache.org/releases/javadoc/current/index.html?org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.html) interface (Java).
```

Comment:
For Python, this should probably link to `GoogleCloudOptions` for consistency - https://beam.apache.org/releases/pydoc/current/apache_beam.options.pipeline_options.html#apache_beam.options.pipeline_options.GoogleCloudOptions